src/webapp/github.py

```diff
@@ -88,35 +151,202 @@ def hit_merge_button(self, owner, repo, num, sha, title=None, msg=None):
+        return self.github_api_call('GET', url, data)
+
+    def get_branch(self, owner, repo, branch):
+        """https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#get-a-branch"""
+        api_path = "/repos/:owner/:repo/branches/:branch"
+        url = api_path2url(api_path, owner=owner, repo=repo, branch=branch)
+
+        return self.github_api_call("GET", url, None)
+
+    def update_file(self, owner, repo, path, message, content, sha=None, branch=None, committer: GithubUser = None,
+                    author: GithubUser = None):
+        """https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents"""
+        api_path = "/repos/:owner/:repo/contents/:path"
+        url = api_path2url(api_path, owner=owner, repo=repo, path=path)
+        data = {
+            "message": message,
+            "content": content
+        }
+        if sha:        data['sha'] = sha
+        if branch:     data['branch'] = branch
+        if committer:  data['committer'] = committer.to_dict()
+        if author:     data['author'] = author.to_dict()
+
+        return self.github_api_call("PUT", url, data)
+
+    def create_pull(self, owner, repo, title, head, base, body=None, maintainer_can_modify: bool = None,
+                    draft: bool = None, issue: int = None):
+        """https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#create-a-pull-request"""
+        api_path = "/repos/:owner/:repo/pulls"
+        url = api_path2url(api_path, owner=owner, repo=repo)
+        data = {
+            "head": head,
+            "base": base
+        }
+        if title: data['title'] = title
+        if body: data['body'] = body
+        if maintainer_can_modify: data['maintainer_can_modify'] = maintainer_can_modify
+        if draft: data['draft'] = draft
+        if issue: data['issue'] = issue
+
+        return self.github_api_call("POST", url, data)
+
+    def get_repo(self, owner, repo):
+        """https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#get-a-repository"""
+        api_path = "/repos/:owner/:repo"
+        url = api_path2url(api_path, owner=owner, repo=repo)
+
+        return self.github_api_call("GET", url, None)
+
+    def get_user(self):
+        """https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-the-authenticated-user"""
+        api_path = "/user"
+        url = api_path2url(api_path)
+
+        return self.github_api_call("GET", url, None)
+
+    def get_user_email(self):
+        """https://docs.github.com/en/rest/users/emails?apiVersion=2022-11-28#list-email-addresses-for-the-authenticated-user"""
+        api_path = "/user/emails"
+        url = api_path2url(api_path)
+
+        return self.github_api_call("GET", url, None)
+
     def get_api_url(self, url):
         return self.github_api_call('GET', url, None)
 
-    def target_repo(self, owner, repo):
-        return GitHubRepoAPI(self, owner, repo)
```

Comment:
You no likie?

This function is used elsewhere (in webhook_app.py) ... i'm thinking it can stay in - you may even want to try it out.